x-pack/plugins/infra/server/lib/adapters/source_status/elasticsearch_source_status_adapter.ts

```diff
@@ -30,10 +30,10 @@ export class InfraElasticsearchSourceStatusAdapter implements InfraSourceStatusA
         .catch(withDefaultIfNotFound<InfraDatabaseGetIndicesResponse>({})),
     ]);
 
-    return indexMaps.reduce(
-      (indexNames, indexMap) => [...indexNames, ...Object.keys(indexMap)],
-      [] as string[]
-    );
+    return indexMaps.reduce((indexNames, indexMap) => {
+      indexNames.push(...Object.keys(indexMap));
```

Comment:
Out of curiosity, why not replace the spread here with `concat`? 
Is the creation of the new array potentially more expensive than the spread of the `currentValue`?